packages/preferences/src/browser/preference-frontend-contribution.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject, named } from "inversify";
+import { CommandContribution, MenuContribution, Command, MenuModelRegistry, CommandRegistry } from "@theia/core";
+import { UserPreferenceProvider } from "./user-preference-provider";
+import { open, OpenerService, CommonMenus, PreferenceScope, PreferenceProvider } from "@theia/core/lib/browser";
+import { WorkspacePreferenceProvider } from './workspace-preference-provider';
+import { FileSystem } from "@theia/filesystem/lib/common";
+import { UserStorageResource } from "@theia/userstorage/lib/browser";
+
+export namespace PreferenceCommands {
+    export const OPEN_USER_PREFERENCES: Command = {
+        id: 'preferences:open_user',
+        label: 'Open User Preferences'
+    };
+    export const OPEN_WORKSPACE_PREFERENCES: Command = {
+        id: 'preferences:open_workspace',
+        label: 'Open Workspace Preferences'
+    };
+}
+
+@injectable()
+export class PreferencesFrontendContribution implements CommandContribution, MenuContribution {
```

Comment:
please use consistent naming `Preference` or `Preferences`, otherwise the file name does not match the concept